tests/components/totalconnect/test_config_flow.py

```diff
@@ -2,49 +2,75 @@
 from homeassistant import data_entry_flow
 from homeassistant.components.totalconnect.const import DOMAIN
 from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .common import CONFIG_DATA, CONFIG_DATA_NO_USERCODES, USERNAME
 
 from tests.async_mock import patch
 from tests.common import MockConfigEntry
 
-USERNAME = "username@me.com"
-PASSWORD = "password"
-
 
 async def test_user(hass):
-    """Test user config."""
-    # no data provided so show the form
+    """Test user step."""
+    # user starts with no data entered, so show the user form
     result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": SOURCE_USER}
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data=None,
     )
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "user"
 
-    # now data is provided, so check if login is correct and create the entry
+
+async def test_user_show_locations(hass):
+    """Test user locations form."""
+    # user/pass provided, so check if valid then ask for usercodes on locations form
     with patch(
         "homeassistant.components.totalconnect.config_flow.TotalConnectClient.TotalConnectClient"
     ) as client_mock:
         client_mock.return_value.is_valid_credentials.return_value = True
         result = await hass.config_entries.flow.async_init(
             DOMAIN,
             context={"source": SOURCE_USER},
-            data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+            data=CONFIG_DATA_NO_USERCODES,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "locations"
+
+
+async def test_user_all_info(hass):
+    """Test all user info included."""
+    # user/pass/usercodes provided, so check if correct and create the entry
+    with patch(
+        "homeassistant.components.totalconnect.config_flow.TotalConnectClient.TotalConnectClient"
+    ) as client_mock, patch(
+        "homeassistant.components.totalconnect.config_flow.TotalConnectClient.TotalConnectLocation"
+    ) as location_mock:
+        client_mock.return_value.is_valid_credentials.return_value = True
+        location_mock.return_value.set_usercode.return_value = True
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=CONFIG_DATA,
         )
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
 
 
-async def test_import(hass):
-    """Test import step with good username and password."""
+async def test_import_good_credentials(hass):
```

Comment:
The import will never have user codes.

Reply:
Given the component won't support YAML now, I've removed `async_step_import` and related tests.